StellarDotnetSdk/Sep/Sep0001/StellarToml.cs

```diff
@@ -0,0 +1,775 @@
+    ///     Fetches and parses stellar.toml from a domain's well-known location.
+    ///     Automatically constructs the standard stellar.toml URL for the given domain
+    ///     and fetches the content via HTTPS. The standard location is always:
+    ///     `https://DOMAIN/.well-known/stellar.toml`
+    /// </summary>
+    /// <param name="domain">The domain name (without protocol). E.g., "example.com"</param>
+    /// <param name="httpClient">Optional custom HTTP client for testing or proxy configuration</param>
+    /// <param name="httpRequestHeaders">Optional custom HTTP headers to include in the request</param>
+    /// <returns>StellarToml instance containing the parsed stellar.toml data</returns>
+    /// <exception cref="StellarTomlException">Thrown when the stellar.toml file is not found or cannot be parsed</exception>
+    public static async Task<StellarToml> FromDomainAsync(
+        string domain,
+        HttpClient? httpClient = null,
+        Dictionary<string, string>? httpRequestHeaders = null)
+    {
+        return await FromDomainAsync(domain, null, null, httpClient, httpRequestHeaders);
+    }
+
+    /// <summary>
+    ///     Fetches and parses stellar.toml from a domain's well-known location.
+    ///     Automatically constructs the standard stellar.toml URL for the given domain
+    ///     and fetches the content via HTTPS. The standard location is always:
+    ///     `https://DOMAIN/.well-known/stellar.toml`
+    /// </summary>
+    /// <param name="domain">The domain name (without protocol). E.g., "example.com"</param>
+    /// <param name="resilienceOptions">Resilience options for HTTP requests. If null, default retry configuration is used.</param>
+    /// <param name="bearerToken">(Optional) Bearer token in case the server requires it.</param>
+    /// <param name="httpClient">Optional custom HTTP client for testing or proxy configuration. If provided, resilienceOptions and bearerToken are ignored.</param>
+    /// <param name="httpRequestHeaders">Optional custom HTTP headers to include in the request</param>
+    /// <returns>StellarToml instance containing the parsed stellar.toml data</returns>
+    /// <exception cref="StellarTomlException">Thrown when the stellar.toml file is not found or cannot be parsed</exception>
+    public static async Task<StellarToml> FromDomainAsync(
+        string domain,
+        HttpResilienceOptions? resilienceOptions,
+        string? bearerToken,
+        HttpClient? httpClient = null,
+        Dictionary<string, string>? httpRequestHeaders = null)
+    {
+        if (string.IsNullOrWhiteSpace(domain))
+        {
+            throw new ArgumentException("Domain cannot be null or empty", nameof(domain));
+        }
+
+        var uriBuilder = new StringBuilder();
+        uriBuilder.Append("https://");
+        uriBuilder.Append(domain);
+        uriBuilder.Append("/.well-known/stellar.toml");
+        var stellarTomlUri = new Uri(uriBuilder.ToString());
```

Comment:
I'd prefer this instead:
```suggestion
        var stellarTomlUri = new Uri($"https://{domain}/.well-known/stellar.toml");
```